openmc/mgxs_library.py

```diff
@@ -379,200 +437,413 @@ def tabular_legendre(self, tabular_legendre):
-            raise ValueError(msg)
+        self._fission = fission
+
+        if np.sum(self._fission) > 0.0:
+            self._fissionable = True
+
+    @kappa_fission.setter
+    def kappa_fission(self, kappa_fission):
+        check_type('kappa_fission', kappa_fission, np.ndarray,
+                   expected_iter_type=Real)
+        check_value('kappa fission shape', kappa_fission.shape,
+                    self.vector_shape)
+
+        self._kappa_fission = kappa_fission
+
+        if np.sum(self._kappa_fission) > 0.0:
+            self._fissionable = True
 
     @chi.setter
     def chi(self, chi):
-        if not self._use_chi:
-            msg = 'Providing chi when nu_fission already provided as matrix!'
-            raise ValueError(msg)
-        if self._representation is 'isotropic':
-            shape = (self._energy_groups.num_groups,)
-        elif self._representation is 'angle':
-            shape = (self._num_polar, self._num_azimuthal,
-                     self._energy_groups.num_groups)
-        # check we have a numpy list
-        check_type("chi", chi, np.ndarray, expected_iter_type=Real)
-        if chi.shape == shape:
-            self._chi = np.copy(chi)
-        else:
-            msg = 'Shape of provided chi "{0}" does not match shape ' \
-                  'required, "{1}"'.format(chi.shape, shape)
-            raise ValueError(msg)
+        if self._use_chi is not None:
+            if not self._use_chi:
+                msg = 'Providing chi when nu_fission already provided as a' \
+                      'matrix'
+                raise ValueError(msg)
+
+        check_type('chi', chi, np.ndarray, expected_iter_type=Real)
+        check_value('chi shape', chi.shape, self.vector_shape)
+
+        self._chi = chi
+
         if self._use_chi is not None:
             self._use_chi = True
 
     @scatter.setter
     def scatter(self, scatter):
-        if self._representation is 'isotropic':
-            shape = (self.num_orders, self._energy_groups.num_groups,
-                     self._energy_groups.num_groups)
-            max_depth = 3
-        elif self._representation is 'angle':
-            shape = (self._num_polar, self._num_azimuthal, self.num_orders,
-                     self._energy_groups.num_groups,
-                     self._energy_groups.num_groups)
-            max_depth = 5
-        # check we have a numpy list
-        check_iterable_type("scatter", scatter, expected_type=Real,
-                            max_depth=max_depth)
-        if scatter.shape == shape:
-            self._scatter = np.copy(scatter)
-        else:
-            msg = 'Shape of provided scatter "{0}" does not match shape ' \
-                  'required, "{1}"'.format(scatter.shape, shape)
-            raise ValueError(msg)
+        check_type('scatter', scatter, np.ndarray, expected_iter_type=Real,
+                   max_depth=len(scatter.shape))
+        check_value('scatter shape', scatter.shape, self.pn_matrix_shape)
```

Comment:
Oh, no. pn_matrix_shape (see latest push).  pn_matrix_shape contains the order as well.